File: guilib-api/src/main/java/net/developertobi/guilib/api/item/ItemEnchantment.java
package net.developertobi.guilib.api.item;

import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemStack;

public record ItemEnchantment(Enchantment enchantment, int strength, boolean isActive) {

    public void performAction(ItemStack item) {
        if (this.isActive) {
            item.addEnchantment(this.enchantment, this.strength);
        } else {
            item.removeEnchantment(this.enchantment);
        }
    }

    public static ItemEnchantment of(Enchantment enchantment, int strength, boolean isActive) {
        return new ItemEnchantment(enchantment, strength, isActive);
    }
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/item/ItemEnchantment.kt
package net.developertobi.inventorylib.api.item

import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemStack

data class ItemEnchantment(val enchantment: Enchantment, val strength: Int, val isActive: Boolean) {

    fun performAction(item: ItemStack) {
        if (this.isActive) item.addEnchantment(this.enchantment, this.strength)
        else item.removeEnchantment(this.enchantment)
    }

    companion object {
        fun of(enchantment: Enchantment, strength: Int, isActive: Boolean): ItemEnchantment {
            return ItemEnchantment(enchantment, strength, isActive)
        }
    }

}
